### DISCUSSION TITLE
Extract dependency injection system to a separate Python package

### DISCUSSION THREAD

**Discussion by dmtrs** [fastapi/fastapi#9024] [Questions] [answered]
### Description / Idea

I started working with `fastapi` and found the Dependency Injection system really elegant and useful. Now, I wish I could use it in other projects that do not involve a `fastapi` dependency.

My idea is to make the dependency injection as a separate python package that fastapi would include.
 
### Example

```Python
import asyncio

from depends import inject, Depends

async def get_something() -> str:
    return 'test'

@inject
async def main(text: str = Depends(get_something())):
    print(text)
```

### The solution you would like

I can understand this not being a priority or motive for the community of the fastapi, but wanted to get some feedback on this. Please feel free to close the request.

**Comment by tiangolo** [comment, marked as answer]
Thank you! The thing is that the main point of the dependency injection system in FastAPI is to be integrated with FastAPI, to extract the data from requests and integrate that with OpenAPI.

So I'm not sure what would be the right way to put this on an external package.

Nevertheless, @adriangb created https://github.com/adriangb/di, I haven't been able to study and play with it yet, but I suspect it's probably right what you need (I also have to review some proposals he was making to change the dependency injection system in FastAPI, hopefully, I'll be able to get to that soon). :nerd_face: 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.

**Comment by dmtrs** [comment]
Hey @tiangolo, thanks for coming back to this 🙏🏼 

I was able to go through FastAPI code and extract parts of the code. Not production ready but you can see here: https://github.com/dmtrs/dependable
